modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java

```diff
@@ -69,37 +85,121 @@ public void testAddObjectValidationRule() throws Exception {
+			ObjectValidationRuleConstants.OUTPUT_TYPE_PARTIAL_VALIDATION,
+			_VALID_DDM_SCRIPT);
+		_testAddObjectValidationRuleFailure(
+			ObjectValidationRuleConstants.ENGINE_TYPE_DDM,
+			ObjectValidationRuleSettingNameException.NotAllowedName.class,
+			String.format(
+				"The object validation rule setting \"%s\" is not allowed",
+				ObjectValidationRuleSettingConstants.NAME_OBJECT_FIELD_ID),
+			RandomTestUtil.randomString(),
+			Collections.singletonList(
+				_getObjectValidationRuleSetting(
+					ObjectValidationRuleSettingConstants.NAME_OBJECT_FIELD_ID,
+					RandomTestUtil.randomString())),
+			ObjectValidationRuleConstants.OUTPUT_TYPE_FULL_VALIDATION,
+			_VALID_DDM_SCRIPT);
+
+		String value = RandomTestUtil.randomString();
+
+		_testAddObjectValidationRuleFailure(
+			ObjectValidationRuleConstants.ENGINE_TYPE_DDM,
+			ObjectValidationRuleSettingValueException.InvalidValue.class,
+			String.format(
+				"The value \"%s\" of the object validation rule setting " +
+					"\"%s\" is invalid",
+				value,
+				ObjectValidationRuleSettingConstants.NAME_OBJECT_FIELD_ID),
+			RandomTestUtil.randomString(),
+			Collections.singletonList(
+				_getObjectValidationRuleSetting(
+					ObjectValidationRuleSettingConstants.NAME_OBJECT_FIELD_ID,
+					value)),
+			ObjectValidationRuleConstants.OUTPUT_TYPE_PARTIAL_VALIDATION,
+			_VALID_DDM_SCRIPT);
+
 		_testAddObjectValidationRuleFailure(
 			StringPool.BLANK,
 			ObjectValidationRuleEngineException.MustNotBeNull.class,
-			"Engine is null", RandomTestUtil.randomString(), _VALID_DDM_SCRIPT);
+			"Engine is null", RandomTestUtil.randomString(),
+			Collections.emptyList(),
+			ObjectValidationRuleConstants.OUTPUT_TYPE_FULL_VALIDATION,
+			_VALID_DDM_SCRIPT);
 		_testAddObjectValidationRuleFailure(
 			"abcdefghijklmnopqrstuvwxyz",
 			ObjectValidationRuleEngineException.NoSuchEngine.class,
 			"Engine \"abcdefghijklmnopqrstuvwxyz\" does not exist",
-			RandomTestUtil.randomString(), _VALID_DDM_SCRIPT);
+			RandomTestUtil.randomString(), Collections.emptyList(),
+			ObjectValidationRuleConstants.OUTPUT_TYPE_FULL_VALIDATION,
+			_VALID_DDM_SCRIPT);
+		_testAddObjectValidationRuleFailure(
+			ObjectValidationRuleConstants.ENGINE_TYPE_GROOVY,
+			ObjectValidationRuleScriptException.class, "syntax-error",
+			RandomTestUtil.randomString(), Collections.emptyList(),
+			ObjectValidationRuleConstants.OUTPUT_TYPE_FULL_VALIDATION,
+			"import;\ninvalidFields = false;");
 
 		_testAddObjectValidationRuleSuccess(
 			ObjectValidationRuleConstants.ENGINE_TYPE_DDM, _VALID_DDM_SCRIPT);
 		_testAddObjectValidationRuleSuccess(
 			ObjectValidationRuleConstants.ENGINE_TYPE_GROOVY,
 			"import com.liferay.commerce.service.CommerceOrderLocalService;\n" +
 				"invalidFields = false;");
+
+		List<ObjectField> objectFields =
+			_objectFieldLocalService.getObjectFields(
+				_objectDefinition.getObjectDefinitionId(), false);
+
+		ObjectField objectField = objectFields.get(0);
```

Comment:
I was thinking, we can't use this get:
```java
_objectFieldLocalService.fetchObjectField(
			_objectDefinition.getObjectDefinitionId(), "textField");
```
To get the specific field that we want?